modules/opentelemetry/src/main/scala/zio/opentelemetry/tracing/TracingSyntax.scala

```diff
@@ -0,0 +1,64 @@
+package zio.opentelemetry.tracing
+
+import io.opentelemetry.common.AttributeValue
+import io.opentelemetry.context.propagation.HttpTextFormat
+import io.opentelemetry.trace.Span
+import zio.ZIO
+import zio.clock.Clock
+import zio.opentelemetry.tracing.attributevalue.AttributeValueConverter
+
+object TracingSyntax {
+
+  implicit final class OpenTelemetryZioOps[R, E, A](val effect: ZIO[R, E, A]) extends AnyVal {
+
+    /**
+     * @see [[Tracing.spanFrom]]
+     */
+    def spanFrom[C](
+      httpTextFormat: HttpTextFormat,
+      carrier: C,
+      reader: PropagationFormat.Reader[C],
```

Comment:
Generally speaking, I like the idea of having functional Reader and Writer. However, I guess the Java API is meant to be used with a ready made Reader/Writer from a library such as Jaeger. Defining our own trait here forces the user of the library to implement it. Therefore I propose to stick to the Java API here. We could provide convenience methods for particular formats in a later PR, which completely hide the Java API.

Reply:
Good point, I hadn't thought of that use case yet. Indeed it makes sense to follow the Java API here

Reply:
Fixed with https://github.com/zio/zio-telemetry/pull/119/commits/70dadc084f45d3e1f4b548d40c5486a71d477edd